lptree.pas

```diff
@@ -4228,10 +4229,58 @@ function TLapeTree_InternalMethod_Goto.Compile(var Offset: Integer): TResVar;
 
 constructor TLapeTree_InternalMethod_Raise.Create(ACompiler: TLapeCompilerBase; ADocPos: PDocPos = nil);
 begin
-  inherited Create('RaiseException', ACompiler, ADocPos);
+  inherited;
   FForceParam := True;
 end;
 
+function TLapeTree_InternalMethod_Raise.Compile(var Offset: Integer): TResVar;
+
+  function InExceptBlock: Boolean;
+  var
+    Node: TLapeTree_Base;
+  begin
+    Node := Self;
+
+    while (Node.Parent <> nil) do
+    begin
+      if (Node.Parent is TLapeTree_Try) then
+        with Node.Parent as TLapeTree_Try do
+          if (Node = ExceptBody) then
+            Exit(True);
+
+      Node := Node.Parent;
+    end;
+
+    Exit(False);
```

Comment:
I'm not really a fan of this style, not really a biggie, but might be better to keep it consistent with other code and use `Result := false`.

```suggestion
    Result := False;
```